### PR TITLE
Support interactive commands

### PR DESCRIPTION
**Pull request by rbren**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

This is a proposal for how to partially solve https://github.com/All-Hands-AI/OpenHands/issues/3030

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

* Give `pexpect` a much shorter timeout for agent-run commands (e.g. 5s)
* If we hit the timeout, report partial STDOUT to the agent, with `exit_code=-1`
* Tell the agent it that if this happens, it needs to send input to STDIN, or `ctrl+c`

---
**Link of any specific issues this addresses**

Fixes: https://github.com/All-Hands-AI/OpenHands/issues/3030
Fixes: https://github.com/All-Hands-AI/OpenHands/issues/892

### CODE REVIEW

File: openhands/runtime/client/client.py [hunk 1]
```diff
@@ -293,11 +314,23 @@ async def run(self, action: CmdRunAction) -> CmdOutputObservation:
...
-                    timeout=action.timeout,
-                    keep_prompt=action.keep_prompt,
-                )
+                if command == '':
+                    output, exit_code = self._continue_bash(
+                        timeout=SOFT_TIMEOUT_SECONDS,
+                        keep_prompt=action.keep_prompt,
+                        kill_on_timeout=False,
+                    )
+                elif command == 'ctrl+c':
```
Comment (xingyaoww) [inline openhands/runtime/client/client.py:323]: maybe `command.loweR() == 'ctrl+c'`?

**Review by xingyaoww — COMMENTED** [review]
LGTM! But i think we can wait for evaluation in https://github.com/All-Hands-AI/OpenHands/pull/3610 to finish first before get this merged.

File: openhands/runtime/client/client.py (hunk 1 above)
Comment (rbren) [inline openhands/runtime/client/client.py:323]:
```suggestion
                elif command.lower() == 'ctrl+c':
```

File: openhands/runtime/client/client.py [hunk 2]
```diff
@@ -507,6 +540,7 @@ async def execute_action(action_request: ActionRequest):
             return event_to_dict(observation)
         except Exception as e:
             logger.error(f'Error processing command: {str(e)}')
+            logger.exception(e)
```
Comment (enyst) [inline openhands/runtime/client/client.py:543]: Do we want both error and exception here?

**Comment by tobitege** [comment]
@rbren integration tests, at least the `test_ipython` ones, should work again once #3776 has been merged.

**Comment by tobitege** [comment]
- integration tests fixed
- removed duplicate exception logging from execute_action

**Review by tobitege — APPROVED** [review]
LGTM
I tested some commands and OH didn't get stuck. In one case a bash was stuck in a MariaDB console, so I told OH to send SIGINT and it quit.
Feels more practical to work with now, but I only had limited time.

**Pull request merged**